### ISSUE TITLE
How to round now() to next 15 minutes time?

### ISSUE THREAD

**Issue by katerlouis** [iamkun/dayjs#1619]
`const now = dayjs(); // e.g. 'HH:mm' is 17:37`

But what I want to modify the object to snap to `17:30` or `17:45`.

Is there a way to do it? "Rounding" only seems to be a thing in relative time, which is not what I'm looking for.

**Comment by katerlouis** [comment]
Friendly push.

**Comment by tanthammar** [comment]
@katerlouis did you find a solution?

**Comment by cjsewell** [comment]
I created my own ceil function, I think I originally stole it from [moment-round](https://github.com/WebDevTmas/moment-round)

```js
import inst, { PluginFunc, UnitType } from 'dayjs';

declare module 'dayjs' {
    interface Dayjs {
        ceil(unit: UnitType, amount: number): inst.Dayjs;
    }
}
const ceil: PluginFunc = (option, dayjsClass) => {
    dayjsClass.prototype.ceil = function (unit, amount) {
        return this.add(amount - (this.get(unit) % amount), unit).startOf(unit);
    };
};
export default ceil;
```

Can be used like

```js
import dayjs from 'dayjs';
import ceil from './ceil';
dayjs.extend(ceil);

console.log(dayjs().ceil('minutes', 15).format());

```

**Comment by vendramini** [comment]
I've created a version to round (up or down) to the nearest time.

19:52 becomes 19:50
19:53 becomes 19:55
```typescript
const ceil: PluginFunc = (option, dayjsClass) => {
  dayjsClass.prototype.round = function (amount, unit) {
    const mod = this.get(unit as UnitType) % amount;

    if(mod < amount / 2) {
      return this.subtract(mod, unit).startOf(unit);
    }

    return this.add(amount - mod, unit).startOf(unit);
  };
};
```

**Comment by cakkypamucm** [comment]
1) I started of using @cjsewell solution, 
but soon I have a bug, as I think:
<pre>
dayjs'2023-06-29T<b>14:14:14</b>+00:00').ceil(1440, 'minutes').utc().format();
</pre>
result is 2023-06-30T**14:00:00**+00:00', 
but I expected '2023-06-30T**00:00:00**+00:00'

2) So this is my solution based on [stackoverflow](https://stackoverflow.com/a/39637877)
```
const round = (option, dayjsClass, dayjsFactory) => {
    // @see https://stackoverflow.com/a/39637877
    dayjsClass.prototype._round = function (amount, unit, method) {
        const duration = dayjsFactory.duration(amount, unit);
        return dayjsFactory(Math[method](this.valueOf() / duration.as("milliseconds")) * duration.as("milliseconds"));
    };

    dayjsClass.prototype.floor = function (amount, unit) {
        return this._round(amount, unit, "floor");
    };

    dayjsClass.prototype.ceil = function (amount, unit) {
        return this._round(amount, unit, "ceil");
    };

    dayjsClass.prototype.round = function (amount, unit) {
        return this._round(amount, unit, "round");
    };
};
export default round;
```

**Comment by herbievine** [comment]
+1

**Comment by Ken-vdE** [comment]
I've answered this on Stackoverflow: [https://stackoverflow.com/a/79884330/3017716](https://stackoverflow.com/a/79884330/3017716)
TL;DR:
```typescript
const someVar = dayjs().set('minute', Math.round((dayjs().minute() + dayjs().second() / 60) / 15) * 15)
```
